File: 110.high_level_syntax/src/main/java/HighLevelSyntax.java
package javax;

public class HighLevelSyntax {
    private static String buildString(final String one, final String two) {
	return one + " " + two;
    }

    public static void main(String[] args) {
	final String hello = "Hello";
	String world = "World!";

	if(args.length == 1) {
	    world = args[0];
	}

	System.out.println(buildString(hello, world));
    }
}

File: 110.high_level_syntax/src/main/scala/scalax/HighLevelSyntax.scala
package scalax

object HighLevelSyntax extends App {
  def getParameters(args: Array[String]): Option[String] = {
    if (args.length == 1) Some(args(0))
    else None
  }

  def buildString(one: String, two: String): String = s"$one $two"

  val hello = "Hello"
  val world = getParameters(args).getOrElse("World!")

  println(buildString(hello, world))
}
